contrib/objectstore/subscriber/objectstore.go

```diff
@@ -54,45 +76,104 @@ func (s *Subscriber) OnStructMessage(c ws.Speaker, msg *ws.StructMessage) {
 	}
 }
 
-// StoreFlows writes a set of flows to the object storage service
+// StoreFlows store flows in memory, before being written to the object store
 func (s *Subscriber) StoreFlows(flows []*flow.Flow) error {
-	if len(flows) == 0 {
-		return nil
+	endTime := time.Now()
+
+	s.flowsMutex.Lock()
+	defer s.flowsMutex.Unlock()
+
+	// transform flows and save to in-memory arrays, based on tag
+	if flows != nil {
+		for _, fl := range flows {
+			var transformedFlow interface{}
+			if s.flowTransformer != nil {
+				transformedFlow = s.flowTransformer.Transform(fl)
+			} else {
+				transformedFlow = fl
+			}
+			if transformedFlow != nil {
+				flowTag := s.flowClassifier.GetFlowTag(fl)
+				s.flows[flowTag] = append(s.flows[flowTag], transformedFlow)
+			}
+		}
```

Comment:
```
                for _, fl := range flows {
                        	tag := s.flowClassifier.GetFlowTag(fl)
                                if s.flowTransformer != nil {
                                        fl = s.flowTransformer.Transform(fl)
                                        if fl == nil {
                                                continue
                                        }
                                }
				s.flows[tag] = append(s.flows[tag], fl)
                  }
```

Reply:
that won't work:
`subscriber/objectstore.go:91:8: cannot use s.flowTransformer.Transform(fl) (type interface {}) as type *flow.Flow in assignment: need type assertion`

Reply:
What about:

```
                  for _, fl := range flows {
                        	tag := s.flowClassifier.GetFlowTag(fl)
                                newfl := interface{}(fl)
                                if s.flowTransformer != nil {
                                        newfl = s.flowTransformer.Transform(fl)
                                }
				s.flows[tag] = append(s.flows[tag], newfl)
                  }
```

Reply:
I intentionally allowed `Transform` to return nil, to indicate that a flow should be discarded.
This will allow security-advisor to discard the "STARTED" flows.